src/View/Helper/TimeHelper.php

```diff
@@ -65,11 +65,11 @@ protected function _getTimezone($timezone)
      *
      * @param \DateTimeInterface|string|int $dateString UNIX timestamp, strtotime() valid string or DateTime object
      * @param \DateTimeZone|string|null $timezone User's timezone string or DateTimeZone object
-     * @return \Cake\I18n\Time
+     * @return \Cake\I18n\FrozenTime
      */
-    public function fromString($dateString, $timezone = null): Time
+    public function fromString($dateString, $timezone = null): FrozenTime
```

Comment:
This is technically a BC break but I doubt this method is used much, we can just document it in the migration guide. The alternative would be to add a new method which returns a `FrozenTime` instance.

Reply:
I'd rather not add new methods that we'll have to remove very soon in the future. We could document that getting a mutable object is now deprecated and will be removed in 5.0 in both the docblock and in the release notes for 4.3.